homeassistant/components/miele/coordinator.py

```diff
@@ -0,0 +1,93 @@
+"""Coordinator module for Miele integration."""
+
+from __future__ import annotations
+
+import asyncio.timeouts
+from dataclasses import dataclass
+from datetime import timedelta
+import logging
+
+from pymiele import MieleAction, MieleDevice
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .api import AsyncConfigEntryAuth
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+type MieleConfigEntry = ConfigEntry[MieleDataUpdateCoordinator]
+
+
+@dataclass
+class MieleCoordinatorData:
+    """Data class for storing coordinator data."""
+
+    devices: dict[str, MieleDevice]
+    actions: dict[str, MieleAction]
```

Comment:
So for my understanding, what is a device and what is an action and how are these related?

Reply:
They are two different streams. Device is device data, like static data and sensor states. Actions signals what actions that are permitted to send in the current moment. E.g. PowerOn is only accepted if power is off. Basically you get a http-400 error if you send the powerOn command in the wrong moment. The steams are normally updated in direct sequence, but it can also be delays between.